src/imbi/models.py

```
@@ -32,6 +34,22 @@ class Blueprint(pydantic.BaseModel):
     json_schema: Schema
     version: int = 0
 
+    @pydantic.model_validator(mode='after')
+    def generate_and_validate_slug(self) -> typing.Self:
+        """Generate slug from name if not provided and validate it."""
+        if self.slug is None:
+            self.slug = slugify.slugify(self.name)
+
+        # Validate slug format
+        if not self.slug:
+            raise ValueError('Slug cannot be empty')
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)